src/main/kotlin/org/jmailen/gradle/kotlinter/support/Reporters.kt

```diff
@@ -1,43 +1,60 @@
 package org.jmailen.gradle.kotlinter.support
 
 import com.pinterest.ktlint.core.Reporter
-import com.pinterest.ktlint.reporter.checkstyle.CheckStyleReporter
-import com.pinterest.ktlint.reporter.html.HtmlReporter
-import com.pinterest.ktlint.reporter.json.JsonReporter
-import com.pinterest.ktlint.reporter.plain.PlainReporter
-import com.pinterest.ktlint.reporter.sarif.SarifReporter
+import com.pinterest.ktlint.core.ReporterProvider
 import java.io.File
 import java.io.PrintStream
+import java.util.ServiceLoader
 
-/* ktlint-disable enum-entry-name-case */
-internal enum class ReporterType(val fileExtension: String) {
-    checkstyle("xml"),
-    html("html"),
-    json("json"),
-    plain("txt"),
-    sarif("sarif.json"),
+internal enum class ReporterType(val id: String, val fileExtension: String) {
+    Checkstyle(id = "checkstyle", fileExtension = "xml"),
+    Html(id = "html", fileExtension = "html"),
+    Json(id = "json", fileExtension = "json"),
+    Plain(id = "plain", fileExtension = "txt"),
+    Sarif(id = "sarif", fileExtension = "sarif.json"),
+    ;
+
+    companion object {
+
+        fun getById(id: String) =
+            values().firstOrNull { it.id == id } ?: error("Unknown reporter type=$id")
+    }
+}
+
+internal fun reporterPathFor(reporterType: ReporterType, output: File, relativeRoot: File) = when (reporterType) {
+    ReporterType.Checkstyle,
+    ReporterType.Html,
+    ReporterType.Json,
+    ReporterType.Plain,
+    -> output.toRelativeString(base = relativeRoot)
+
+    ReporterType.Sarif -> output.absolutePath
 }
-/* ktlint-enable enum-entry-name-case */
-
-fun reporterFor(reporterName: String, output: File): Reporter {
-    val out = PrintStream(output)
-    return SortedThreadSafeReporterWrapper(
```

Comment:
I removed `SortedThreadSafeReporterWrapper` as it doesn't seem to be needed. It was added when the source files was chunked, which isn't the case anymore 👀 

I'd have to go through repo history to remind myself why chunking got dropped, but given that still sounds like a good idea to me, if we considered restoring it I'd re-visit the approach with `SortedThreadSafeReporterWrapper` and rather create a separate task that would do the reporting (to ensure correct output and avoid parallelization issues). 
From my perspective, we're now changing _correctness_ and the focus on performance could be the next step.